### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by frm**

### CODE REVIEW

File: lib/dictator.ex [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+
+  # lib/my_app_web/router.ex
+  defmodule MyAppWeb.Router do
+    pipeline :authorised do
+      plug Dictator
+    end
+  end
+  ```
+
+  Requires Phoenix (or at least `conn.private[:phoenix_action]` to be set).
```
Comment (naps62) [inline lib/dictator.ex:25]: not sure what this line is refering to. I'm using it at the router level, and without setting this

File: lib/dictator.ex [hunk 2]
```diff
@@ -1,4 +1,58 @@
...
+  To load resources from the database, requires Ecto. See `Dictator.Policies.EctoSchema`.
+
+  Dictator assumes your policies are in `lib/my_app_web/policies/` and follow
```
Comment (naps62) [inline lib/dictator.ex:28]: the directory path isn't really true, only the module namespace, right?

File: lib/dictator.ex [hunk 3]
```diff
@@ -1,4 +1,58 @@
...
+  * `policy`: policy to apply. See above to understand how policies are inferred.
+  * `key`: key under which the current user is placed in `conn.assigns`. Defaults to `:current_user`.
+
+  ## Configuration options
+
+  Options that you can place in your `config/*.exs` files.
+
+  * `key`: Same as the `:key` parameter in the plug option section. The plug option takes precedence, meaning you can place it in a config and then override it in specific controllers or pipelines.
+  * `unauthorized_handler`: Handler to be called when the user is not authorised to access the resource. Defaults to `Dictator.UnauthorizedHandlers.Default`.
+  """
```
Comment (naps62) [inline lib/dictator.ex:54]: I'll have to later check what this looks like on hexdocs, but I was expecting the moduledoc to be a bit less in-depth, and then separate pages could deal with the different examples and configurations

File: lib/dictator.ex [hunk 4]
```diff
@@ -36,12 +90,12 @@ defmodule Dictator do
         nil
       end
 
-    if apply(policy, :can?, [user, action, %{params: conn.params, resource: resource}]) do
+    params = %{params: conn.params, resource: resource, opts: opts}
```
Comment (naps62) [inline lib/dictator.ex:93]: maybe opts? having a `:params` within a `params` looks confusing

Comment (frm) [reply lib/dictator.ex:93]: I'm also adding opts here

File: lib/dictator.ex (hunk 3 above)
Comment (frm) [reply lib/dictator.ex:54]: You can clone this and run `mix docs && open docs/index.html`

File: lib/dictator.ex (hunk 2 above)
Comment (frm) [reply lib/dictator.ex:28]: Yeah, but isn't this the "Elixir-way"?

File: lib/dictator.ex (hunk 1 above)
Comment (frm) [reply lib/dictator.ex:25]: Hm, yeah, fair. I'll remove it

File: lib/dictator.ex (hunk 2 above)
Comment (naps62) [reply lib/dictator.ex:28]: it is, but this package doesn't assume that in any way, which is what your sentence says

Comment (frm) [reply lib/dictator.ex:28]: How we about "we recommend placing them in"?

**Review by naps62 — APPROVED** [review]

**Pull request merged**